### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by sinkuu**

### CODE REVIEW

**Comment by lzutao** [comment]
cc @alexcrichton (for rust build)

File: src/librustc_codegen_ssa/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,6 @@ test = false
 bitflags = "1.0.4"
 cc = "1.0.1"
 num_cpus = "1.0"
-rustc-demangle = "0.1.15"
```
Comment (fbstj) [inline src/librustc_codegen_ssa/Cargo.toml]: is `rustc-demangle` not intentionally for some reason?

Comment (mati865) [reply src/librustc_codegen_ssa/Cargo.toml]: Looks like it was never used, cc @eddyb https://github.com/rust-lang/rust/commit/b9e5cf99a9acd9c29d02c2f27c0eb223fd0a92be

Comment (eddyb) [reply src/librustc_codegen_ssa/Cargo.toml]: Copy-pasted but the only use is still in `rustc_codegen_llvm`.

**Pull request merged**